testing/api/test_api.py

```diff
@@ -1408,11 +1766,103 @@ def test_update_profile(testrun, reset_profiles, add_profile): # pylint: disable
   # Check if profile was updated
   assert updated_profile_check is not None
 
+def test_update_profile_invalid_json(testrun, reset_profiles, add_profile): # pylint: disable=W0613
+  """Test for update profile invalid JSON payload (no 'name')"""
+
+  # Load the new profile using add_profile fixture
+  add_profile("new_profile.json")
+
+  # Load the updated profile using load_json utility method
+  updated_profile = load_json("profile_invalid_format.json",
+                              directory="testing/api/profiles")
+
+  # Send the post request to update the profile
+  r = requests.post(
+      f"{API}/profiles",
+      data=json.dumps(updated_profile),
+      timeout=5)
+
+  # Parse the response
+  response = r.json()
+
+  # Check if status code is 400 (Bad request)
+  assert r.status_code == 400
+
+  # Check if "error" key in response
+  assert "error" in response
+
+def test_create_profile_invalid_json(testrun, reset_profiles): # pylint: disable=W0613
+  """Test for create profile invalid JSON payload """
+
+  # Load the profile using load_json utility method
+  new_profile = load_json("profile_invalid_format.json",
+                          directory="testing/api/profiles")
+
+  # Send the post request to update the profile
+  r = requests.post(
+      f"{API}/profiles",
+      data=json.dumps(new_profile),
+      timeout=5)
+
+  # Parse the response
+  response = r.json()
+
+  # Check if status code is 400 (Bad request)
+  assert r.status_code == 400
+
+  # Check if "error" key in response
+  assert "error" in response
+
+  # Load the 2nd profile
+  new_profile_2 = {"name": "New Profile"}
+
+  # Send the post request to update the profile
+  r = requests.post(
+      f"{API}/profiles",
+      data=json.dumps(new_profile_2),
+      timeout=5)
+
+  # Parse the response
+  response = r.json()
+
+  # Check if status code is 400 (Bad request)
+  assert r.status_code == 400
+
+  # Check if "error" key in response
+  assert "error" in response
+
+@responses.activate
+def test_create_update_profile_internal_server_error(testrun): # pylint: disable=W0613
+  """Test for create/update profile causing internal server error."""
+
+  # Mock the POST request for create/update profiles API response
+  responses.add(
```

Comment:
Find the scenario when this will occur and test that rather than mocking the response.